lib/langchain/llm/openai.rb

```diff
@@ -104,52 +114,14 @@ def complete(prompt:, **params)
     #
     # @param messages [Array<Hash>] List of messages comprising the conversation so far
     # @param model [String] ID of the model to use
-    def chat(
-      messages: [],
-      model: defaults[:chat_completion_model_name],
-      frequency_penalty: nil,
-      logit_bias: nil,
-      logprobs: nil,
-      top_logprobs: nil,
-      max_tokens: nil,
-      n: defaults[:n],
-      presence_penalty: nil,
-      response_format: nil,
-      seed: nil,
-      stop: nil,
-      stream: nil,
-      temperature: defaults[:temperature],
-      top_p: nil,
-      tools: [],
-      tool_choice: nil,
-      user: nil,
-      &block
```

Comment:
@codenamev You didn't like how all of the params were listed out? We had that previously because I felt like it made the interface clearer since it's an interface that we control.

Reply:
I was torn here, but allowing these inherited classes to change the method signature with custom keyword arguments complicates things quite a lot and is adding confusion on when to use what.  One thing we could do is change this to accept a `UnifiedParameters` instance (instead of a Hash as it is) and then pass that along to the `chat_parameters` the same way (and just have it merge).  Doing that would allow us to continue documenting and inspect what is acceptable criteria (similar to how GraphQL mutations can accept input types).

Reply:
What would the DX look like then? Something along the lines of:
```ruby
uniparams = Langchain::LLM::UnifiedParameters.new schema: {}, parameters: params
openai.chat parameters: uniparams
```
?

Reply:
Just to re-iterate, the goal of these changes is that `Langchain::LLM::Base` is the source of truth for the unified method signature we want applied to all children.  Preserving this idea, the LLM sub-classes will be responsible for:

1. Adhering to the unified method signature of its' parent
2. Managing any amendments to the method signature it wants to handle

The way this currently is implemented, each sub-class uses a `chat_params` instance from `LLM::Base` and modifies it [in its' initializer](https://github.com/patterns-ai-core/langchainrb/pull/600/files#diff-e9b54ad7b6ef063bedb31fba426cea87e7b816d4eeaa65a4ffb07929e6e58646R43-R52).

Following `3` of OpenRouter's principles here:

> If the chosen model doesn't support a request parameter, then the parameter will be ignored. The rest will be forwarded to the underlying model API.

The callers should be able to pass _any_ parameters to these sub-classes and the model itself will know how to handle them for their respective API calls.  For this reason, this is why I modified the arguments of `chat` for OpenAI to accept a hash rather than explicit keyword arguments.  Under the hood, the _model_ forces the unified schema, ignores any fields, re-maps (aliases) any fields it needs re-named, and allows any additional fields the model accepts.

In the future, we can add validations and parameter inquiries, but this is mostly a small step backward (obfuscating the method signature) in favor of applying the unified signature with backwards compatibility.  I'll be adding further documentation for what these parameters are that I think will solve some of your worries here 😄

Reply:
@codenamev Okay, I love it, and I'm on board 🚢 🚀

Reply:
(A proper docs website is desperately needed) but we'd probably still want to catalogue all of the possible values that could be passed under the `params={}` hash [here](https://github.com/patterns-ai-core/langchainrb/pull/600/files#diff-e9b54ad7b6ef063bedb31fba426cea87e7b816d4eeaa65a4ffb07929e6e58646R114-R116) so that it shows up in the [rubydoc](https://rubydoc.info/gems/langchainrb/Langchain/LLM/OpenAI#chat-instance_method) as well.